File: Arrays/Tournament Winner/solution1.py
def tournamentWinner(competitions, results):
    # Write your code here.
    currentBest = ""
    scores = {currentBest: 0}

    for i,compi in enumerate(competitions):
        result = results[i]
        home, away = compi
        win = home if result==1 else away
        updateScore(win,3,scores)
        if scores[win] > scores[currentBest]:
            currentBest = win

    return currentBest

def updateScore(team,points,scores):
    if team not in scores:
        scores[team] = 0
    scores[team] += points

File: Arrays/Tournament Winner/solution1.cpp
#include <vector>
#include <algorithm>
#include <unordered_map>
#include <unordered_set>

using namespace std;

string tournamentWinner(vector<vector<string>> competitions,
                        vector<int> results) 
{

      string currentBest = "";
      unordered_map<string, int> scores= {{currentBest,0}};
      for(int i =0; i< competitions.size(); i++){
            auto result = results[i];
            auto competition = competitions[i];
            auto home = competition[0];
            auto away = competition[1];
            
            auto winner =  result == 1 ? home : away;
            updateScore(winner,3,scores);

            if(scores[winner] > scores[currentBest])
                  currentBest = winner;
      }
      return currentBest;
}

void updateScore(string team,int points,unordered_map<string,int> &scores){
      if (scores.find(team) == scores.end()){
            scores[team] = 0;
      }
      scores[team] += points;
}